inbox/api/update.py

```diff
@@ -9,26 +9,28 @@
 # STOPSHIP(emfree): better naming/structure for this module
 
 
-def update_message(message, request_data, db_session):
+def update_message(message, request_data, db_session, optimistic=True):
     accept_labels = message.namespace.account.provider == 'gmail'
     # Update flags (message.{is_read, is_starred})
     unread, starred = parse_flags(request_data)
-    update_message_flags(message, db_session, unread, starred)
+    update_message_flags(message, db_session, unread, starred,
+                         optimistic=optimistic)
     # Update folders/ labels (message.categories)
     if accept_labels:
         labels = parse_labels(request_data, db_session, message.namespace_id)
         if labels is not None:
             added_labels = labels - set(message.categories)
             removed_labels = set(message.categories) - labels
             update_message_labels(message, db_session, added_labels,
-                                  removed_labels)
+                                  removed_labels, optimistic=optimistic)
     else:
         folder = parse_folder(request_data, db_session, message.namespace_id)
         if folder is not None:
-            update_message_folder(message, db_session, folder)
+            update_message_folder(message, db_session, folder,
+                                  optimistic=optimistic)
```

Comment:
(The above would also be nice, because it'd ensure we never forget to do this.) Leaving off `optimistic=optimistic` would cause the wrong behavior here and would be very subtle.